### PR TITLE
Response Caching

### LINKED ISSUE #733: Response Caching
- Intro to response caching
  - Specify how you want to cache response messages- 
- Response Cache Attribute
  - A filter that specifies what HTTP cache headers to set
  - Duration, location (anyclient/none), vary by header value, no store
    - Any = proxy or client
    - Client = client only
    - None - no caching
  - Multiple named cache profiles can be configured on MvcOptions and specified using the CacheProfileName property on the attribute
    - Analog to output cache profiles in System.Web
  - Attribute settings win over cache profile
- Server caching behavior
  - IIS, WebListener will do kernel mode caching
  - Typically IIS is used as a reverse proxy, so you will need to specify Any for the location
  - No caching support in Kestrel, but Kestrel is typically run behind a reverse proxy that could do caching.

### PR DESCRIPTION
**Pull request by ardalis**
Fixes #733.
@pranavkm @Tratcher please review.

### CODE REVIEW

File: mvc/performance/response-caching.rst [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+`View or download sample from GitHub <https://github.com/aspnet/Docs/tree/1.0.0-rc1/mvc/performance/response-caching/sample>`_.
+
+What is Response Caching
+------------------------
+*Response caching* refers to specifying cache-related headers on HTTP responses made by ASP.NET MVC actions. These headers specify how you want client and intermediate (proxy) machines to cache responses to certain requests (if at all). This can reduce the number of requests a client or proxy makes to the web server, since future requests for the same action may be served from the client or proxy's cache. In this case, the request is never made to the web server.
+
+.. image:: response-caching/_static/proxy-and-cache.png
+
+The primary HTTP header used for caching is ``Cache-Control``. The `HTTP specification <https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html>`_ details many options for this directive. Three common directives are:
```
Comment (Tratcher) [inline mvc/performance/response-caching.rst]: https://tools.ietf.org/html/rfc7234 is a more current reference.

Comment (ardalis) [inline mvc/performance/response-caching.rst]: Updated both links.

**Comment by Tratcher** [comment]
Seems ok, waiting on @pranavkm.

File: mvc/performance/response-caching.rst [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+Produces the following headers:
+
+.. code-block:: javascript
+
+	Cache-Control: public,max-age=60
+
+Cache Profiles
+^^^^^^^^^^^^^^
+Instead of duplicating ``ResponseCache`` settings on many controller actions, cache profiles can be configured as options when setting up MVC in the ``ConfigureServices`` method in ``Startup``. Values found in a referenced cache profile will be used as the defaults by the ``ResponseCache`` attribute, and will be overridden by any properties specified on the attribute.
```
Comment (pranavkm) [inline mvc/performance/response-caching.rst]: > on many ResponseCache attributes

Comment (ardalis) [inline mvc/performance/response-caching.rst]: fixed.

**Comment by pranavkm** [comment]
Looks :shipit: to me.

**Pull request merged**